school.rb

```diff
@@ -0,0 +1,11 @@
+class School
+  attr_reader :age, :phase, :salary, :target_raise

```

Comment:
A school has an age, phase etc? I would expect a school to have students, teacher, etc.  Use naming that makes sense to say "A student is a (kind of) X" and "A teacher is a (kind of) X".  I'd suggest `Person`.

Also, since you're using this as a superclass for both students and teachers, it should not include salary and target raise.  Just include the things that ALL of your subclasses need.